src/protocols/secure_channel/PASESession.cpp

```diff
@@ -776,6 +785,12 @@ CHIP_ERROR PASESession::ValidateReceivedMessage(ExchangeContext * exchange, cons
     }
 
     VerifyOrReturnError(!msg.IsNull(), CHIP_ERROR_INVALID_ARGUMENT);
+
+    if (payloadHeader.HasMessageType(mLastProcessedMsg))
+    {
+        return CHIP_ERROR_DUPLICATE_MESSAGE_RECEIVED;
+    }
```

Comment:
This is supposed to be prevented by underlying layers. Add a TODO to remove once fixed

Reply:
As above, defensive coding wise, why would this be removed?

Reply:
Are all protocols doing this defensive coding? Should we add all possible "defensive coding" where it actually means "add code that should not be there because the underlying layers are not finished"?